lib/httpDigest.js

```diff
@@ -1,55 +1,100 @@
-import base64 from './base64.js';
+import {encode as base64Encode} from 'universal-base64';
+import {encode as base64urlEncode} from 'base64url-universal';
 import crypto from './crypto.js';
 
 /**
- * Creates a string suitable for HTTP Digest: header.
+ * Creates a value suitable for the HTTP `Digest` header.
  *
- * @param {string|object} data - Objects to be hashed (typically, headers).
- * @param {string} algorithm - Hash algorithm to use. (e.g. 'sha256').
- * @param {boolean} useMultihash - Whether to encode via multihash.
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - Objects to be hashed
+ *   (typically headers).
+ * @param {string} [options.algorithm] - Hash algorithm to use.
+ *   (e.g. 'sha256').
+ * @param {boolean} options.useMultihash - Whether to encode via multihash.
  *   If false, the hash will be base64 encoded (non-url).
  *
- * @returns {Promise<string>} Resolves with hashed digest string.
+ * @returns {Promise<string>} Resolves to `Digest` header value.
  */
-async function create({data, algorithm = 'sha256', useMultihash = true}) {
+export async function createHeaderValue(
+  {data, algorithm = 'sha256', useMultihash = true} = {}) {
+  const {key, encodedDigest} = await _createHeaderValueComponents(
+    {data, algorithm, useMultihash});
+  return `${key}=${encodedDigest}`;
+}
+
+/**
+ * Verifies the HTTP `Digest` header value against the given HTTP body `data`.
+ *
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - The data to be verified.
+ * @param {string} options.headerValue - The digest header value to verify
+ *   the data against.
+ *
+ * @returns {object} - An object with `verified` set to `true` or `false`.
+ */
+export async function verifyHeaderValue({data, headerValue} = {}) {
+  const {key, algorithm, encodedDigest} = _parseHeaderValue(headerValue);
+  const {encodedDigest: expectedDigest} = await _createHeaderValueComponents(
+    {data, algorithm, useMultihash: key === 'mh'});
+  return {verified: encodedDigest === expectedDigest};
+}
+
+async function _createHeaderValueComponents(
+  {data, algorithm = 'sha256', useMultihash = true} = {}) {
   if(typeof data !== 'string') {
     data = JSON.stringify(data);
   }
-  const encodedData = new TextEncoder().encode(data);
-  let digest;
+  if(algorithm !== 'sha256') {
+    throw new Error(`Algorithm "${algorithm}" is not supported.`);
+  }
+  const digest = await _getDigest({data, algorithm});
+  if(useMultihash) {
+    return {key: 'mh', encodedDigest: _createMultihash({digest})};
+  }
+  return {key: 'SHA-256', encodedDigest: base64Encode(digest)};
+}
 
-  switch(algorithm) {
-    case 'sha256': {
-      digest = new Uint8Array(
-        await crypto.subtle.digest({name: 'SHA-256'}, encodedData));
-      break;
-    }
-    case 'hs2019': {
-      // todo: support hs2019 / SHA-512
+function _parseHeaderValue(headerValue) {
+  const [key, ...rest] = headerValue.split('=');
```

Comment:
I'm trying to imagine what the data format coming in here is.  Can you provide a sample?

Reply:
@mattcollier Here's what the data format looks like.


```
key: SHA-256
rest: X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE
```

Reply:
@JSAssassin thank you.  I made a replit to try and make sense of this parsing algorithm: https://repl.it/@mattcollier/FavoriteUnfoldedDemand#index.js

is the output `X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=` (with a trailing equal) the expected value of `encodedDigest` on L63?

Reply:
@mattcollier from my understanding, that's correct. The expected value of encodedDigest should have a trailing equal.

Note: updated how rest looks in my previous comment(made an error earlier)

Reply:
In that case, I would like to offer this alternative implementation that I think is easier to reason about: https://repl.it/@mattcollier/alternateSplit

Reply:
Updated - 4db2f55